sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_control_flow_pipeline.py

```diff
@@ -44,6 +54,86 @@ class TestConditionalNodeInPipeline(AzureRecordedTestCase):
     pass
 
 
+class TestIfElse(TestConditionalNodeInPipeline):
+    def test_happy_path_if_else(self, client: MLClient, randstr: Callable[[], str]) -> None:
```

Comment:
May combine test_happ_path_if_else, test_if_else_one_branch and test_if_else_one_branch into one test, using `@pytest.mark.parametrize`.

Reply:
Currently in e2e test, each parametrized test case will become a file with long file name, which need to take more steps to handle. Reference `test_pipeline_job_with_command_job` in `pipeline_job/e2etests` as an example. I think split tests makes it clearer compared to use the fixture.